packages/core/src/conflict-resolution.ts

```diff
@@ -190,32 +194,24 @@ export async function fetchLog(
   const commit = await dispatcher.retrieveCommit(cid)
   if (!commit) throw new Error(`No commit found for CID ${cid.toString()}`)
 
-  let payload = commit
-  const isSignedCommit = StreamUtils.isSignedCommit(commit)
-  if (isSignedCommit) {
-    payload = await dispatcher.retrieveCommit(commit.link)
-    if (!payload) throw new Error(`No commit found for CID ${commit.link.toString()}`)
-  }
-  const prevCid: CID = payload.prev
-  if (!prevCid) {
-    // this is a fake log
-    return []
-  }
+  let prevCid: CID = commit.prev
```

Comment:
Previous version of the code ensured `prevCid` is really a `CID`, this one makes it `CID | undefined` until line 203. Fishy. Is it necessary to do it this way? Types are quite important for understanding code.

Reply:
My thinking was that even in the earlier code, if this was a "fake log" as the comment says (I don't know what that means) or a `GenesisCommit` (?), that `prevCid` could still be `undefined` and so this logic essentially doesn't change the spirit of the code. I can definitely rewrite this though.

Reply:
Maybe you could wait to create `prevCid` until after building `nextCommitData`, then take `nextCommitData.commit.prev`?

Reply:
Good idea!

Reply:
And I think I finally see what @ukstv was trying to say 🤦🏼